reframe/utility/jsonext.py

```diff
@@ -38,19 +59,8 @@ def encode(obj):
     if inspect.istraceback(obj):
         return traceback.format_tb(obj)
 
-    return None
-
-
-def _make_json_friendly(obj):
-    '''Fallback type converter if json dump(s) raises a TypeError.'''
-
-    _valid_keys = {str, int, float, bool, type(None)}
-    if isinstance(obj, MutableMapping):
-        if not all((any(isinstance(k, t) for t in _valid_keys) for k in obj)):
-            newobj = type(obj)()
-            for k, v in obj.items():
-                newobj[util.toalphanum(str(k))] = v
-
+    newobj = encode_dict(obj)
+    if newobj:
```

Comment:
Better check explicitly for None here:

```suggestion
    if newobj is not None:
```